src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -508,6 +500,32 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
                 default:
                     throw ExceptionUtilities.UnexpectedValue(syntax.Kind());
             }
+
+            void reportNullableReferenceTypesIfNeeded(SyntaxToken questionToken, TypeSymbolWithAnnotations typeArgument = default)
+            {
+                if (InExecutableBinder)
+                {
+                    // Inside a method body or other executable code, we can pull on NonNullTypes symbol or question IsValueType without causing cycles.
+                    // We still need to delay that check when binding in an attribute argument
+                    if (!ShouldCheckConstraintsNullability)
+                    {
+                        diagnostics.Add(new LazyMissingNonNullTypesContextDiagnosticInfo(Compilation, NonNullTypesContext, typeArgument), questionToken.GetLocation());
+                    }
```

Comment:
Please combine these checks to reduce the number of code paths:
```
if (!InExecutable || !ShouldCheckConstraintsNullability)
{
   diagnostics.Add(new LazyMissingNonNullTypesContextDiagnosticInfo(...));
}
else
{
   ...
}
```

Reply:
> Please combine these checks to reduce the number of code paths:

Suggested refactoring will require duplicating condition checks

---
In reply to: [217495231](https://github.com/dotnet/roslyn/pull/29809#discussion_r217495231) [](ancestors = 217495231)

Reply:
> Please combine these checks to reduce the number of code paths:

I will follow up on this in the next PR

---
In reply to: [217506307](https://github.com/dotnet/roslyn/pull/29809#discussion_r217506307) [](ancestors = 217506307,217495231)